### PR TITLE
Completed SGT-Back-End Challenge

### PR DESCRIPTION
**Pull request by alvarodiazjr**
Screen Shot 2022-08-25 at 2 53 14 PM

Screen Shot 2022-08-25 at 3 02 12 PM

Screen Shot 2022-08-25 at 3 50 03 PM

Screen Shot 2022-08-25 at 3 50 30 PM

Screen Shot 2022-08-25 at 4 01 10 PM

Screen Shot 2022-08-26 at 10 05 49 AM

Screen Shot 2022-08-26 at 10 06 19 AM

Screen Shot 2022-08-26 at 10 06 35 AM

Screen Shot 2022-08-26 at 10 07 18 AM

Screen Shot 2022-08-26 at 10 07 36 AM

Screen Shot 2022-08-26 at 10 07 49 AM

### CODE REVIEW

File: sgt-back-end/index.js [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+    })
+    .catch(err => {
+      console.error(err);
+      res.status(500).json({ error: 'An unexpected error occured' });
+    });
+});
+
+app.post('/api/grades', (req, res) => {
+  const input = req.body;
+  const [name, course, score] = [input.name, input.course, Number(input.score)];
```
Comment (TimelikeClosure) [inline sgt-back-end/index.js]: Any particular reason you're creating an array, then immediately destructuring it and throwing it away?

File: sgt-back-end/index.js [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+  if (!name || !course || !score) {
+    res.status(400).json({ error: 'Must include Name, Course, and Score.' });
+  } else if (!Number.isInteger(score) || score < 1 || score > 100) {
+    res.status(400).json({ error: 'Score must be a positive integer between 1 and 100' });
```
Comment (TimelikeClosure) [inline sgt-back-end/index.js]: A `0` score shouldn't trigger a `400` status code

**Review by TimelikeClosure — CHANGES_REQUESTED** [review]

**Comment by alvarodiazjr** [comment]
Screen Shot 2022-08-26 at 11 20 55 AM
Screen Shot 2022-08-26 at 11 21 33 AM

File: sgt-back-end/index.js [hunk 3]
```diff
@@ -0,0 +1,133 @@
...
+  const course = input.course;
+  const score = input.score;
+  if (score < 0 || score > 100 || Number.isNaN(score)) {
+    res.status(400).json({ error: 'Score must be a positive integer between 0 and 100' });
+    return;
+  }
+  if (!name || !course || !score) {
+    res.status(400).json({ error: 'Must include Name, Course, and Score.' });
+    return;
+  }
```
Comment (TimelikeClosure) [inline sgt-back-end/index.js]: A non-integer `score` in the `0` to `100` range should still result in a `400 Bad Request` (ex: `43.297`).

File: sgt-back-end/index.js [hunk 4]
```diff
@@ -0,0 +1,133 @@
...
+app.put('/api/grades/:gradeId', (req, res) => {
+  const gradeId = Number(req.params.gradeId);
+  const input = req.body;
+  const name = input.name;
+  const course = input.course;
+  const score = input.score;
+  if (!Number.isInteger(gradeId) || gradeId < 0) {
+    res.status(400).json({ error: 'invalid gradeId input' });
+    return;
+  }
```
Comment (TimelikeClosure) [inline sgt-back-end/index.js]: A grade id of `0` should also be treated as invalid (database ids start at `1`).

**Review by TimelikeClosure — CHANGES_REQUESTED** [review]

File: sgt-back-end/index.js [hunk 5]
```diff
@@ -0,0 +1,127 @@
...
+  returning *
+  `;
+  const value = [gradeId];
+  db.query(sql, value)
+    .then(result => {
+      const grade = result.rows[0];
+      if (!grade) {
+        res.status(404).json({ error: 'cannot find grade with gradeId of ' + gradeId });
+      } else {
+        res.status(204).json(grade);
```
Comment (TimelikeClosure) [inline sgt-back-end/index.js]: Your `204 No Content` responses should not have content.

**Review by TimelikeClosure — CHANGES_REQUESTED** [review]

**Comment by alvarodiazjr** [comment]
Screen Shot 2022-08-26 at 1 05 05 PM

**Review by TimelikeClosure — APPROVED** [review]

**Pull request merged**